cmake/aikidoConfig.cmake.in

```diff
@@ -0,0 +1,138 @@
+set(aikido_VERSION x.y.z)
+
+@PACKAGE_INIT@
+
+function(aikido_traverse_components output_variable)
+  set(output_components)
+
+  foreach(component ${ARGN})
+    # We should be able to elminate the <C>_VISITED property by detecting
+    # whether the <C>_FOUND property is DEFINED. Unfortunately, DEFINED checks
+    # whether or not define_property() was called, not whether the variable was
+    # set. The exact behavior seems to vary between CMake versions, so we use
+    # an extra variable instead.
+    get_property(is_visited DIRECTORY PROPERTY "aikido_${component}_VISITED")
+    set_property(DIRECTORY PROPERTY "aikido_${component}_VISITED" TRUE)
+
+    if(NOT is_visited)
+      set(component_include
+        "${CMAKE_CURRENT_LIST_DIR}/aikido_${component}Component.cmake")
+      set(target_include
+        "${CMAKE_CURRENT_LIST_DIR}/aikido_${component}Targets.cmake")
+
+      if(EXISTS "${component_include}" AND EXISTS "${target_include}")
+        include("${component_include}")
+
+        set_property(DIRECTORY PROPERTY "aikido_${component}_FOUND" TRUE)
```

Comment:
These are intended to be parent-scoped output variables. I am storing them in `DIRECTORY` properties because:
- `PARENT_SCOPE` only works for one level of scope
- using a `macro` clobbers all local variables (e.g. `component`) during recursion
- a `GLOBAL` property will pollute strictly more scope

The ideal solution would be to use a `PARENT_SCOPE` variable and _very carefully_ forward all of the variables recursively up the recursion stack.

Another option is to use `GLOBAL` or `CACHE`d properties, but prefix them to make it clear that they're an implementation detail not intended for external use (e.g. `aikido_internal_`). Otherwise, setting these variables defeats the purpose of carefully setting `CACHE`d properties for _only the components that the user asks for_.

In any case: this is an implementation detail. I'd prefer to change this in another pull request.